### PR TITLE
golangci-lint integration

### PR DESCRIPTION
**Pull request by azdagron**
Fixes up all of the errors in the codebase and adds a linting step to Travis CI.

Most of the fixes should be trivial. Some judiciousness was taken on errcheck failures. Errors from operations were generally ignored when:
1. the operations were inherently safe but returned an error due to interface conformance (e.g. io.Writer interfaces writing to memory buffers)
1. Errors would be otherwise unreportably (e.g. HTTP response writer failure, CLI failure to write to stderr)
1. server shutdown code in tests that wasn't directly related to the testing, but only if checking the error would have required too much refactoring

### CODE REVIEW

**Comment by azdagron** [comment]
This PR is only intended to introduce the linter and make it happy. I'd be all for another PR making a pass to fix acronym usage according to the Go naming guidelines. We'd want to come up with a good set of rules. For example, in some places, it might make more sense to say "WorkloadID" than "SPIFFEID" and so on. Also, `CSRSPIFFEID` is a little unwieldy so we'd probably want to come up with something nice there. Protobuf definitions and generated code of course will need exceptions since `protoc-gen-go` has its own rules for converting field names.

**Comment by amartinezfayo** [comment]
@azdagron I agree with that.

**Comment by azdagron** [comment]
Thanks for the review @amartinezfayo!

Let's hold off merging this until we have a heads up from @evan2645, who is prepping a 0.9.1 release.

File: pkg/server/ca/manager_test.go [hunk 1]
```diff
@@ -468,7 +467,7 @@ func (s *ManagerSuite) TestRunFailsIfNotifierFails() {
 	s.m = NewManager(s.selfSignedConfig())
 	s.setNotifier(fakenotifier.New(fakenotifier.Config{
 		OnNotifyAndAdvise: func(req *notifier.NotifyAndAdviseRequest) (*notifier.NotifyAndAdviseResponse, error) {
-			return nil, errors.New("OH NO!")
+			return nil, errors.New("ohno")
```
Comment (evan2645) [inline pkg/server/ca/manager_test.go]: OH YEAH

Comment (azdagron) [reply pkg/server/ca/manager_test.go]: LOL.

File: pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go [hunk 2]
```diff
@@ -265,7 +268,7 @@ func (s *AttestorSuite) TestAttestFailsIfTokenSignatureCannotBeVerifiedByCluster
 func (s *AttestorSuite) TestAttestSuccess() {
 	// Success with FOO signed token (local validation)
 	token := s.signToken(s.fooSigner, "NS1", "SA1")
-	status := createTokenStatus("NS1", "SA1", true)
+	createTokenStatus("NS1", "SA1", true)
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go]: You are right. Will remove.

File: pkg/common/util/task.go [hunk 3]
```diff
@@ -28,9 +28,7 @@ func RunTasks(ctx context.Context, tasks ...func(context.Context) error) error {
 	runTask := func(task func(context.Context) error) (err error) {
 		defer func() {
 			if r := recover(); r != nil {
-				errch <- fmt.Errorf("panic: %v\n%s\n", r, string(debug.Stack()))
-			} else {
-				errch <- err
+				err = fmt.Errorf("panic: %v\n%s\n", r, string(debug.Stack())) //nolint: golint // newlines are intentional
```
[the remark this answers is not in the document]
Comment (azdagron) [reply pkg/common/util/task.go]: It is reported. The defer changes the `err` value being returned from the runTask closure, which is sent over the `errCh` goroutine.

**Comment by amartinezfayo** [comment]
@azdagron Have you considered to have `lint` as part of the `make all` target? I'm thinking that it may be useful to make sure that build passes with tests + lint before submitting a PR.

**Comment by azdagron** [comment]
That's a decent thing to do. Is it ok if we integrate that into the new makefile changes (#1311)?

**Comment by amartinezfayo** [comment]
> That's a decent thing to do. Is it ok if we integrate that into the new makefile changes (#1311)?

Sure, that makes sense.

**Pull request merged**